### ISSUE TITLE
ST7789_Sample does not seem to run on a Raspberry Pi Zero 2 W

### ISSUE THREAD

**Issue by DocBrown101** [WildernessLabs/Meadow.Linux#20]
Hello, I am trying to get the ST7789_Sample to run, unfortunately without success.

I always get this exception:
``
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'o')
   at Meadow.ServiceCollection.Add[TRegisterAs](TRegisterAs o)
   at Meadow.App`1..ctor()
   at MeadowTftDemo.MeadowApp..ctor() in /root/git/Experimental-Projects/MeadowTftDemo/MeadowTftDemo/MeadowApp.cs:line 19
   at MeadowTftDemo.Program.Main(String[] args) in /root/git/Experimental-Projects/MeadowTftDemo/MeadowTftDemo/Program.cs:line 17
``

Unfortunately I can't get more details out, can you please give me a hint what I might have done wrong?

I just started it with ``dotnet run`` in the project folder. (.NET SDKs installed: 6.0.403)

**Comment by ctacke** [comment]
Not 100% sure on this specific error - I can't reproduce it.  I also am running on a Pi 4B, though I would expect it to be the same.

The sample has not been updated to match the RC1 app lifecycle, and this can be a little confusing.  Your Meadow "application" is no longer the application/entry point.  Instead, the entry point is now `Meadow.dll` and your code needs to be compiled as a Library.  Think of your code as a "plug-in" that Meadow.dll will load.

I've made the code changes in the branch here (basically removing Program.cs and changing the compile type):
https://github.com/WildernessLabs/Meadow.Linux/tree/bug/sample-updates/src/samples/pi/ST7789_Sample

The primary change, however, is that to run your application, it is now:

```
$ dotnet Meadow.dll
```

**Comment by DocBrown101** [comment]
Thank you for the quick reply. That makes it understandable now!

But what was still unclear to me, how do I get the Meadow.dll when I include it via nuget.
I have solved it this way. I have added the following two lines to my csproj file:
``
<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
<GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
``

After that I could start the build on the raspberry pi zero:
``dotnet build --configuration Release``

In the directory bin/Release/net6.0 the file Meadow.runtimeconfig.json had to be present, so that I could use ``dotnet Meadow.dll`` without errors.

But now it runs without problems, thank you!
